sys/parsoid.js

```diff
@@ -344,7 +392,9 @@ class ParsoidService {
                         body: res.body[format].body
                     };
                     resp.headers.etag = mwUtil.makeETag(rp.revision, tid);
-                    return this.saveParsoidResult(hyper, req, format, tid, res)
+                    return this.saveParsoidResultToLatest(hyper, req, tid, res)
+                    .catch({ status: 412 }, () =>
```

Comment:
> The best I can come up with using existing conditions would be if-none-match: W/"12345": Only store if existing content does not weakly match rev 12345. I guess another option would be to create a new header, for example if-version-lt: 12345.

The problem with this idea is that on this level we do not know the value of "12345" It's only know inside the `parsoid_bucket`...

> For some of these more specialized buckets we could also consider calling them directly as modules, rather than making HTTP calls to them through the router. 

The `parsed_bucket` is indeed kinda a submodule of `parsed` module, so library-style call could make a lot of sense here. But we are going to generalize the `parsed_bucket` to reuse it elsewhere, in that sense the library-style call is worse... 

Maybe replace 412 with 400? Like "You're trying to put the latest stuff but the revId is not actually latest so it's illegal"?